services/environment-pruner/prune.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# set -e -o pipefail
+
+# Create a JWT admin token to talk to the API.
+API_ADMIN_JWT_TOKEN=$(./create_jwt.sh)
+BEARER="Authorization: bearer $API_ADMIN_JWT_TOKEN"
+
+CUTOFF_DATE=$(date +'%Y-%m-%d %H:%M:%S' --date '-90 days')
+
+# Load all the projects.
+GRAPHQL='query environments {
+  environments:allProjects {
+    name
+    productionEnvironment
+    openshift {
+      consoleUrl
+      token
+      name
+    }
+    environments {
+      openshiftProjectName
+      name
+      updated
+    }
+  }
+}'
+
+# Convert GraphQL file into single line (but with still \n existing), turn \n into \\n, esapee the Quotes
+query=$(echo $GRAPHQL | sed 's/"/\\"/g' | sed 's/\\n/\\\\n/g' | awk -F'\n' '{if(NR == 1) {printf $0} else {printf "\\n"$0}}')
+ALL_ENVIRONMENTS=$(curl -s -XPOST -H 'Content-Type: application/json' -H "$BEARER" api:3000/graphql -d "{\"query\": \"$query\"}")
+
+echo "$ALL_ENVIRONMENTS" | jq -c '.data.environments[] | select((.environments|length)>=1)' | while read project
+  do
+    PROJECT_NAME=$(echo "$project" | jq -r '.name')
+    OPENSHIFT_URL=$(echo "$project" | jq -r '.openshift.consoleUrl')
+    PRODUCTION_BRANCH=$(echo "$project" | jq -r '.productionEnvironment')
+
+    echo "$project" | jq -c '.environments[]' | while read environment
+    do
+      ENVIRONMENT_OPENSHIFT_PROJECTNAME=$(echo "$environment" | jq -r '.openshiftProjectName')
+      ENVIRONMENT_NAME=$(echo "$environment" | jq -r '.name')
+      UPDATED_DATE=$(echo "$environment" | jq -r '.updated')
+
+      echo "$OPENSHIFT_URL - $PROJECT_NAME: handling environment $ENVIRONMENT_NAME"
+
+      if [ "$ENVIRONMENT_NAME" = "$PRODUCTION_BRANCH" ]; then
+        # This should only ever remove non-prods.
+        echo "$PROJECT_NAME: $ENVIRONMENT_NAME is the production environment"
+        continue;
+      fi
+
+      Skip if no openshift project exists.
```

Comment:
should be a comment yeah?

Reply:
Whoops! Yeah - I went a bit crazy with my uncommenting while I was testing.